src/hazelcore/dynamics/Dynamics.re

```diff
@@ -2115,6 +2142,8 @@ module Evaluator = {
     | IntLit(_)
     | FloatLit(_)
     | Triv => BoxedValue(d)
+    | FailedAssert => Indet(FailedAssert)
```

Comment:
Actually this seems to suggest that failed assertions evaluate to indet rather than unit, which would change my suggestion above. Is this correct?

Reply:
failed assertions have unknown type and are indet. only successful assertions evaluate to triv.